### PR TITLE
Jobs view to search jobs #514

### PR DESCRIPTION
**Pull request by pcaston**
1.  Renamed index.html.haml as company_jobs.html.haml and search_jobs.html.haml as  index.html.haml.
2.  Added hide/show toggle for jobs search form.
3.  'Click on any column title to sort.' informative message added to jobs list when number of jobs returned from search is > 1.
4.  Added list company jobs to footer for company persons, agency admin, job developers and case workers.

### CODE REVIEW

File: spec/controllers/jobs_controller_spec.rb [hunk 1]
```diff
@@ -236,6 +214,38 @@
       it_behaves_like 'unauthorized', 'visitor'
     end
   end
+
```
Comment (pcaston) [inline spec/controllers/jobs_controller_spec.rb]: removed

File: config/routes.rb [hunk 2]
```diff
@@ -127,8 +127,8 @@
   # ------------------------------ Jobs ----------------------------------------
   get 'jobs/list/:job_type'         => 'jobs#list',   as: :list_jobs
   get 'jobs/:job_id/apply/:user_id' => 'jobs#apply',  as: :apply_job
-  get 'jobs/list_search_jobs'       => 'jobs#list_search_jobs',
-      as: :list_search_jobs
+  get 'jobs/list_company_jobs'       => 'jobs#company_jobs',
```
Comment (pcaston) [inline config/routes.rb]: removed

File: app/views/jobs/index.html.haml [hunk 3]
```diff
@@ -1,21 +1,172 @@
...
+                All companies with one or more
+                %i active
+                jobs are listed.
+                Click in the search box and select a company from the list that appears.
+                %br
+                You can select multiple companies by repeatedly clicking
+                in the search box.
+
+      .modal-footer
+        %button.btn.btn-default(type='button' data-dismiss='modal') Close
```
Comment (pcaston) [inline app/views/jobs/index.html.haml]: No change from previous.

File: app/views/jobs/index.html.haml [hunk 4]
```diff
@@ -1,21 +1,172 @@
...
+        %i job matches are performed using all search fields which you have set.
+        For example, if you specify that the job title should contain all of
+        the words "Project Manager" and have selected the job skill
+        "Budget Development" then only jobs that meet
+        %i both
+        of those criteria will be shown.
+        %br
+        %br
+
+        %table.table.table-hover(style='margin-bottom: 0px;')
```
Comment (pcaston) [inline app/views/jobs/index.html.haml]: No change from previous.

File: app/views/jobs/index.html.haml [hunk 5]
```diff
@@ -1,21 +1,172 @@
...
+
+
+= render partial: 'searched_job_list'
+
+#search_help.modal.fade(role='dialog' tabindex='-1')
+  .modal-dialog.modal-lg
+    .modal-content
+      .modal-header
+        %button.close(type='button' class='close' data-dismiss='modal')
+          %span(aria-hidden='true') &times;
```
Comment (pcaston) [inline app/views/jobs/index.html.haml]: No change from previous.

File: app/views/jobs/index.html.haml [hunk 6]
```diff
@@ -1,21 +1,172 @@
...
+      %hr
+
+
+= render partial: 'searched_job_list'
+
+#search_help.modal.fade(role='dialog' tabindex='-1')
+  .modal-dialog.modal-lg
+    .modal-content
+      .modal-header
+        %button.close(type='button' class='close' data-dismiss='modal')
```
Comment (pcaston) [inline app/views/jobs/index.html.haml]: No change from previous.

File: app/views/jobs/index.html.haml [hunk 7]
```diff
@@ -1,21 +1,172 @@
...
+      %hr
+    %td(style='width: 1px; padding: 0 10px; white-space: nowrap; font-weight: bold;')
+      = "#{pluralize(@jobs.count, 'Job')} Found." + "#{sort_instruction(@jobs.count)}"
+    %td
+      %hr
+
+
+= render partial: 'searched_job_list'
+
+#search_help.modal.fade(role='dialog' tabindex='-1')
```
Comment (pcaston) [inline app/views/jobs/index.html.haml]: No change from previous.

File: app/views/jobs/index.html.haml [hunk 8]
```diff
@@ -1,21 +1,172 @@
...
+.row
+  .col-sm-3
+    %h2 Search Jobs
+.row
+  .col-sm-3
+    %a(id='toggle_search_form' href='#job_search_form' class='text-danger'
+                    style='padding-left: 10px;')
+      Show Search Form
+
+#job_search_form(style = 'display: none;')
```
Comment (pcaston) [inline app/views/jobs/index.html.haml]: No change from previous.

File: app/views/jobs/index.html.haml [hunk 9]
```diff
@@ -1,21 +1,172 @@
...
-.col-sm-offset-4.col-sm-6
-  = will_paginate @jobs
+- provide(:title, 'Jobs Search')
+
+.row
+  .col-sm-3
+    %h2 Search Jobs
+.row
+  .col-sm-3
+    %a(id='toggle_search_form' href='#job_search_form' class='text-danger'
```
Comment (pcaston) [inline app/views/jobs/index.html.haml]: No change from previous.

File: app/helpers/jobs_helper.rb [hunk 10]
```diff
@@ -2,4 +2,8 @@ module JobsHelper
   def list_jobs
     @jobs ||= Job.take(25)
   end
+
```
Comment (pcaston) [inline app/helpers/jobs_helper.rb]: removed

**Comment by pcaston** [comment]
Applied additional semaphore fixes

**Comment by pcaston** [comment]
Fixed jobs status checks

**Comment by pcaston** [comment]
Closing to correct jobs_controller_spec.rb tests

**Comment by pcaston** [comment]
Re-running semaphore tests after correction

**Comment by pcaston** [comment]
Fixed some hound issues

**Comment by pcaston** [comment]
Re-open to rerun tests

**Comment by patmbolger** [comment]
Regarding the change to the footer (good idea, by the way) ...
1. These items show up not only for company person but for an agency person as well:

screen shot 2017-02-07 at 6 26 42 am
This should show up _only_ for company person.

2. For the company person, lets put up just these additional items in the footer:
  - `List Jobs`
  - `Post Job` (note singular job)

(remove `Search resumes` - they actually have to go to a particular job to do that anyway)

File: app/controllers/jobs_controller.rb [hunk 11]
```diff
@@ -260,7 +261,7 @@ def notify_job_developer
 
     # Parameters: {"job_developer_id"=>"3", "company_person_id"=>"1",
     #              "job_seeker_id"=>"3", "id"=>"202"}
-
+
```
Comment (pcaston) [inline app/controllers/jobs_controller.rb]: removed

File: app/controllers/jobs_controller.rb [hunk 12]
```diff
@@ -78,14 +74,19 @@ def create
...
       set_company_address
       render :new
     end
   end
 
+  def company_jobs
+    @jobs = policy_scope(Job).order(:title).includes(:company)
+                             .paginate(page: params[:page], per_page: 20)
+  end
+
```
Comment (pcaston) [inline app/controllers/jobs_controller.rb]: removed

**Comment by patmbolger** [comment]
@pcaston - please rebase this PR (merge conflict noted above).

File: features/step_definitions/jobs_steps.rb [hunk 13]
```diff
@@ -25,18 +25,9 @@
 Then(/^I should( not)? see "Revoked" next to "([^"]*)"$/) do |negate, job_title|
 	@job ||= Job.find_by(title: job_title)
 	if negate
-		expect(page).not_to have_css("#job-#{@job.id} span", text: "Revoked")
+		expect(page).not_to have_selector(:xpath, "//tr[td[contains(.,'#{job_title}')]]/td/span", :text => 'Revoked')
```
Comment (pcaston) [inline features/step_definitions/jobs_steps.rb]: replaced tab

File: features/step_definitions/jobs_steps.rb [hunk 14]
```diff
@@ -25,18 +25,9 @@
...
-	end
-end
-
-Then(/^I should( not)? see "revoke" button for "([^"]*)"$/) do |negate, job_title|
-	@job ||= Job.find_by(title: job_title)
-	if negate
-		expect(page).not_to have_css("#job-#{@job.id} button", text: "revoke")
-	else
-		expect(page).to have_css("#job-#{@job.id} button", text: "revoke")
+		expect(page).to have_selector(:xpath, "//tr[td[contains(.,'#{job_title}')]]/td/span", :text => 'Revoked')
```
Comment (pcaston) [inline features/step_definitions/jobs_steps.rb]: replaced tab

File: app/views/layouts/_footer.html.haml [hunk 15]
```diff
@@ -1,18 +1,16 @@
...
+    %div{ :class => "footer-left" }
+      %span= "&reg;".html_safe+" MetPlus #{Date.today.year}"
+    %div{ :class => "footer-right" }
+      - if pets_user.is_a?(CompanyPerson)
+        = link_to "Post job", new_job_path(company_id: current_user,
+        company_person_id: current_user.try(:actable_id)),
+        class: "#{yield(:new_job_active)}"
+        |
+      - elsif pets_user.try(:agency)  && pets_user.is_job_developer?(pets_user.try(:agency))
+        = link_to "Post job", new_job_path,
```
Comment (pcaston) [inline app/views/layouts/_footer.html.haml]: removed

File: app/views/layouts/_footer.html.haml [hunk 16]
```diff
@@ -1,18 +1,16 @@
...
+        company_person_id: current_user.try(:actable_id)),
+        class: "#{yield(:new_job_active)}"
+        |
+      - elsif pets_user.try(:agency)  && pets_user.is_job_developer?(pets_user.try(:agency))
+        = link_to "Post job", new_job_path, 
+        class: "#{yield(:new_job_active)}"
+        |
 
-            #{ link_to "Partner with Us", new_company_registration_path, class: yield(:new_company_active) }
+      #{ link_to "Partner with Us", new_company_registration_path, class: yield(:new_company_active) }
```
Comment (pcaston) [inline app/views/layouts/_footer.html.haml]: can't continue on next line

File: features/step_definitions/jobs_steps.rb [hunk 17]
```diff
@@ -24,20 +24,13 @@
...
-end
-
-Then(/^I should( not)? see "revoke" button for "([^"]*)"$/) do |negate, job_title|
-	@job ||= Job.find_by(title: job_title)
-	if negate
-		expect(page).not_to have_css("#job-#{@job.id} button", text: "revoke")
-	else
-		expect(page).to have_css("#job-#{@job.id} button", text: "revoke")
-	end
+  if negate
```
Comment (pcaston) [inline features/step_definitions/jobs_steps.rb]: corrected

File: features/step_definitions/jobs_steps.rb [hunk 18]
```diff
@@ -24,20 +24,13 @@
...
+      expect(page).not_to have_selector(:xpath, "//tr[td[contains(.,
```
Comment (pcaston) [inline features/step_definitions/jobs_steps.rb]: corrected

File: features/step_definitions/jobs_steps.rb [hunk 19]
```diff
@@ -24,20 +24,13 @@
...
+      '#{job_title}')]]/td/span", :text => 'Revoked')
```
Comment (pcaston) [inline features/step_definitions/jobs_steps.rb]: corrected

File: features/step_definitions/jobs_steps.rb [hunk 20]
```diff
@@ -24,20 +24,13 @@
...
+  else
+      expect(page).to have_selector(:xpath, "//tr[td[contains(.,
```
Comment (pcaston) [inline features/step_definitions/jobs_steps.rb]: corrected

File: features/step_definitions/jobs_steps.rb [hunk 21]
```diff
@@ -24,20 +24,13 @@
...
+      '#{job_title}')]]/td/span", :text => 'Revoked')
```
Comment (pcaston) [inline features/step_definitions/jobs_steps.rb]: corrected

File: features/step_definitions/jobs_steps.rb (hunk 20 above)
Comment (pcaston) [inline features/step_definitions/jobs_steps.rb]: corrected

File: features/step_definitions/jobs_steps.rb (hunk 21 above)
Comment (pcaston) [inline features/step_definitions/jobs_steps.rb]: can't be wrapped on commas

File: app/views/layouts/_footer.html.haml [hunk 22]
```diff
@@ -1,18 +1,16 @@
...
+  %div{ :class => :container }
+    %div{ :class => 'footer-left' }
+      %span= "&reg;".html_safe+" MetPlus #{Date.today.year}"
+    %div{ :class => "footer-right" }
+      - if pets_user.is_a?(CompanyPerson)
+        = link_to 'Post job', new_job_path(company_id: current_user,
+        company_person_id: current_user.try(:actable_id)),
+        class: "#{yield(:new_job_active)}"
+        |
+      - elsif pets_user.try(:agency) && pets_user.is_job_developer?(pets_user.try(:agency))
```
Comment (pcaston) [inline app/views/layouts/_footer.html.haml]: no way to break up

File: app/views/jobs/_searched_job_list.html.haml [hunk 23]
```diff
@@ -1,33 +1,40 @@
...
+          = highlight(truncate(job.description, length: 200), @description_words)
+        %td
+          = job.company.name
+        %td
+          = job.shift
+        %td
+          -if job.address
+            = job.address.state
+        %td
+          -if job.address
```
Comment (pcaston) [inline app/views/jobs/_searched_job_list.html.haml]: fixed

File: app/views/jobs/_searched_job_list.html.haml [hunk 24]
```diff
@@ -1,33 +1,40 @@
...
+            = job.address.city
+        %td
+          -unless job.active?
```
Comment (pcaston) [inline app/views/jobs/_searched_job_list.html.haml]: fixed

**Comment by pcaston** [comment]
Closing and Re-openning pull request to re-run semaphore tests.

**Comment by pcaston** [comment]
Re-open to re-run semaphore tests.

**Pull request merged**